src/ucp/tag/tag_recv.c

```diff
@@ -77,7 +78,7 @@ ucp_tag_recv_common(ucp_worker_h worker, void *buffer, size_t count,
         ucp_recv_desc_release(rdesc);
 
         if (req_flags & UCP_REQUEST_FLAG_CALLBACK) {
-            cb(req + 1, status, &req->recv.tag.info);
+            cb(req + 1, status, &req->recv.tag.info, req->recv.tag.user_data);
```

Comment:
do we have some flag to prevent/allow calling completion callback from recv function?

Reply:
i mean, that if recv is completed return OK and not call the callback

Reply:
we probably need to return sender tag and length from function itself, like we do for stream receive.
Maybe add it to ucp_request_param_t (either by value or by pointer)?
In some use cases, it's confusing that completion callback is called from the function itself, and we can fix this as part of new request API.